.github/workflows/syft-pr_tests.yml

```diff
@@ -67,76 +72,25 @@ jobs:
           pip install "pillow>=8.3.1"
 
       - name: Scan for security issues
+        if: steps.changes.outputs.backend == 'true'
         run: |
           bandit -r src
           safety check
 
       - name: Install libomp for XGBoost
-        if: startsWith(runner.os, 'macos')
+        if: startsWith(runner.os, 'macos') && steps.changes.outputs.backend == 'true'
         run: |
           brew install libomp
           pip uninstall python-dp -y
 
       - name: Run normal tests
+        if: steps.changes.outputs.backend == 'true'
         run: |
           pytest -m 'fast or slow' --cov syft --cov-fail-under 70 -n auto --durations=50
```

Comment:
This is one of the trick , I used in ``SyMPC`` repo for speeding up CI,
https://github.com/OpenMined/SyMPC/blob/634396a9d663eda514f66654a96c23ae2ab7d2de/.github/workflows/tests.yml#L189
Instead of setting auto , we could set a custom like ``8``,  The value 8 really worked for our repo, we could also experiment with the values, I am sure, It would greatly speed the CI

Reply:
Sweet stuff! @rasswanth-s thanks for the tip, I'll have to test that and compare then open another PR for it. Also, how did you compare between auto & a value like 8 to find the magnitude of the difference? Would love to know. Also I know that `auto` checks for available system resource before assigning a process, so a fixed number like `8` makes a classical assumption that there will be always resource for 8 workers. Are there supposed downsides to this?

Reply:
Sure @bitsofsteve , the runners in Github CI have two cores for ``ubuntu`` and three cores ``mac os``, when there was ``auto`` only two and three workers were set by default, My intution was that , increasing workers ,would make it more multi threaded,it worked sort of trial and error. 

One **downside : was that , when I set high number workers(more than 8) , it frequently crashed the workers, we should like kind of set an optimal number.
I remember windows runner having problem with 8 workers, maybe we could start with ``6`` and check?

Reply:
Using more cores than there are CPUs would only be a good idea if there are a lot of non CPU bound tests. It might have been in the SyMPC repo that you had a lot of waiting going on between tests. I think for now `auto` is fine since we are running a lot of different tests across the Syft repo most of which are doing work, but I would be interested to see with some benchmarks if we can get things faster by tweaking stuff like this.